Sources/SPMBuildCore/BuildParameters/BuildParameters+Testing.swift

```diff
@@ -117,10 +132,11 @@ extension BuildParameters {
             // when building and testing in release mode, one can use the '--disable-testable-imports' flag
             // to disable testability in `swift test`, but that requires that the tests do not use the testable imports feature
             self.enableTestability =  enableTestability ?? (.debug == configuration)
-            self.testProductStyle = targetTriple.isDarwin() ? .loadableBundle : .entryPointExecutable(
+            self.testProductStyle = (targetTriple.isDarwin() && library == .xctest) ? .loadableBundle : .entryPointExecutable(
                 explicitlyEnabledDiscovery: forceTestDiscovery,
                 explicitlySpecifiedPath: testEntryPointPath
             )
+            self.library = library
```

Comment:
could be nice to somehow unify the flows such that in all cases we generate a derived executable product and in the case of Darwin xctest it does the "load bundle" as an implementation detail

Reply:
I don't disagree, but it's beyond the scope of this particular PR.